pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -373,9 +383,18 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation, operationType
 			Fn:           flow.TaskFn(botanist.RestartControlPlanePods).DoIf(restartControlPlaneControllers),
 			Dependencies: flow.NewTaskIDs(deployKubeControllerManager, deployControlPlane, deployControlPlaneExposure),
 		})
-		f = g.Compile()
 	)
 
+	for k, v := range botanist.Shoot.Components.DNS.AdditionalProviders {
+		_ = g.Add(flow.Task{
+			Name:         fmt.Sprintf("Ensuring additional DNSProvider %q", k),
+			Fn:           flow.TaskFn(component.OpWaiter(v).Deploy).DoIf(!o.Shoot.HibernationEnabled),
+			Dependencies: flow.NewTaskIDs(waitUntilKubeAPIServerServiceIsReady),
```

Comment:
```suggestion
			Dependencies: flow.NewTaskIDs(deployInternalDomainDNSRecord, deployExternalDomainDNSRecord, deployIngressDomainDNSRecord),
```

Best effort to let the internal/external provider create the internal/external records (dns records do not have a ref spec to a provider).